### ISSUE TITLE
Notifications not removed upon board restriction/membergroup change

### ISSUE THREAD

**Issue by jasonclemons** [SimpleMachines/SMF#2987]
Per this topic: http://www.simplemachines.org/community/index.php?topic=539013.0

Notifications are not being removed for restricted boards after membergroup change.

Also affects 2.0.x, and quite possibly 1.1.x

**Comment by Oldiesmann** [comment]
I'll look into this when I get home from work in a few hours. If I remember correctly, SMF does check board access when determining who to notify about new replies. If that's the case, then really the only way this could happen is the notification email got added to the mail queue before the user was removed from the group (in which case there may not be much we can do about it, but I don't remember how info for the mail queue is stored in the database).

**Comment by illori** [comment]
when i have been removed from the team group i have not received email notifications for topics i have on watch.

**Comment by margarett** [comment]
As I stated in the linked topic, this cannot happen in a clean install unless, as Oldies said, the notification is already in the queue. 
I'd say there's nothing to fix here ;)

**Comment by Oldiesmann** [comment]
I did some further checking and it turns out this **can** happen (at least in 2.0 - still looking into things for 2.1), but only in a specific situation (the member cannot have any additional (non-primary) groups and the board must be set so that no other groups have access to it besides admins). I've posted a fix for that (and a related issue) for 2.0 in the topic. I'll commit a fix for 2.1 if I can verify that this is still an issue in 2.1 as well.
